worksheet_D/oxo.py

```diff
@@ -0,0 +1,108 @@
+
+class OxoBoard:
+    def __init__(self):
+        #Initialises board array
+        self.gameBoard = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+
+
+    def get_square(self, x, y):
+        #Returns 0, 1 or 2 depending on the contents of the specified square.
+        return  self.gameBoard[x][y]
+
+
+    def set_square(self, x, y, mark):
+        #Checks content of square and fills empty spots with marks and returns True, is space is taken returns False
+        if self.gameBoard[x][y] == 0:
+            self.gameBoard[x][y] = mark
+            return True
+        else:
+            return False
+
+
+    def is_board_full(self):
+        #Checks every square to see if there are any empty squares left, returns True if none are empty
+        for x in xrange(3):
+            for y in xrange(3):
+                if self.gameBoard[x][y] == 0:
+                    return False
+        return True
+
+
+    def get_winner(self):
+        # Checks whole board to see if there are 3 in row (vertical, horizontal and diagonal)
+        for i in xrange(3):
+            if self.gameBoard[i][0] == self.gameBoard[i][1] == self.gameBoard[i][2] and self.gameBoard[i][0] != 0:
+                return self.gameBoard[i][0]
+            elif self.gameBoard[0][i] == self.gameBoard[1][i] == self.gameBoard[2][i] and self.gameBoard[0][i]  != 0:
+                return self.gameBoard[0][i]
+            elif self.gameBoard[0][0] == self.gameBoard[1][1] == self.gameBoard[2][2] and self.gameBoard[0][0] != 0:
```

Comment:
The diagonal checks don't need to be inside the `for` loop -- this results in checking the diagonals three times